src/policy/mallocspace/metadata.rs

```diff
@@ -1,61 +1,137 @@
-use atomic::Ordering;
-
-use crate::util::constants;
-use crate::util::conversions;
-use crate::util::heap::layout::vm_layout_constants::BYTES_IN_CHUNK;
+use crate::util::heap::layout::vm_layout_constants::{BYTES_IN_CHUNK, LOG_BYTES_IN_CHUNK};
 use crate::util::metadata::load_metadata;
 use crate::util::metadata::side_metadata;
+#[cfg(target_pointer_width = "64")]
+use crate::util::metadata::side_metadata::metadata_address_range_size;
+#[cfg(target_pointer_width = "32")]
+use crate::util::metadata::side_metadata::metadata_bytes_per_chunk;
 use crate::util::metadata::side_metadata::SideMetadataContext;
 use crate::util::metadata::side_metadata::SideMetadataSpec;
-#[cfg(target_pointer_width = "64")]
+use crate::util::metadata::side_metadata::GLOBAL_SIDE_METADATA_BASE_ADDRESS;
 use crate::util::metadata::side_metadata::LOCAL_SIDE_METADATA_BASE_ADDRESS;
 use crate::util::metadata::store_metadata;
+use crate::util::metadata::MetadataSpec;
 use crate::util::Address;
 use crate::util::ObjectReference;
+use crate::util::{constants, conversions, metadata};
 use crate::vm::{ObjectModel, VMBinding};
-
-use std::collections::HashSet;
-use std::sync::RwLock;
+use std::sync::atomic::Ordering;
+use std::sync::Mutex;
 
 lazy_static! {
-    pub static ref ACTIVE_CHUNKS: RwLock<HashSet<Address>> = RwLock::default();
+    pub(super) static ref CHUNK_METADATA: SideMetadataContext = SideMetadataContext {
+        global: metadata::extract_side_metadata(&[MetadataSpec::OnSide(
+            ACTIVE_CHUNK_METADATA_SPEC
```

Comment:
`ACTIVE_CHUNK_METADATA_SPEC` is always on-side.
So, you can directly write `global: vec![ACTIVE_CHUNK_METADATA_SPEC]` here

Reply:
Are you still happy for me to use the `CHUNK_METADATA` for eagerly mmaping the active chunk space? I understand it is using it incorrectly, but I could not find another way of doing the eager mmaping.

Reply:
I guess we don't have other choices. You have to map metadata for addresses that we/mmtk haven't mapped.